File: go/problems/day4_test.go
package problems

import "testing"

var day4 Solvable[int] = Day4{}

func TestDay4_TestSet_ShouldYield_Result(t *testing.T) {
	input := readProblemInput("day4_1")
	result := day4.Solve(input)

	expected := 13
	if result != expected {
		t.Errorf("expected %d but got %d", expected, result)
	}
}

func TestDay4_FullSet_ShouldYield_Result(t *testing.T) {
	input := readProblemInput("day4_2")
	result := day4.Solve(input)

	expected := 17803
	if result != expected {
		t.Errorf("expected %d but got %d", expected, result)
	}
}

File: ts/problems/day4.test.ts
import test, { describe } from "node:test";
import assert from "node:assert";

import { solve } from "./day4.js";
import { readProblemInput } from "../inputReader.js";

describe("Day 4", () => {
  test("test set should yield expected result", () => {
    const input = readProblemInput("day4_1");
    const result = solve(input);
    assert.equal(result, 13);
  });

  test("full set should yield expected result", () => {
    const input = readProblemInput("day4_2");
    const result = solve(input);
    assert.equal(result, 17803);
  });
});
